### PR TITLE
Boundary checks for coordinates in GeoJSON

### LINKED ISSUE #218: Query fails when GeoJSON polygon is not WGS84
Hello all,

I am running following script that used to work just a month ago:
```python
footprint = geojson_to_wkt(read_geojson('C:\DATA_raw\shapefiles\Fields\Field011_32633.geojson'))
products = api.query(footprint,date=('20180401', '20180417'),producttype='S2MSI2Ap',
                 platformname = 'Sentinel-2',cloudcoverpercentage = (0, 100),)
```

Unfortunately, I am getting following error:
`TypeError: int() argument must be a string or a number, not 'NoneType'`
that follows another error:

`File "C:\Users\Marcela\Anaconda2\envs\PESSL_27\lib\site-packages\sentinelsat\sentinel.py", line 287, in _load_subquery
    total_results = int(json_feed['opensearch:totalResults'])`

Why are the totalResults None?

thanks a lot for your help,
Marcela

### PR DESCRIPTION
**Pull request by Andrey-Raspopov**
Closes #218 and #236

### CODE REVIEW

**Comment by Andrey-Raspopov** [comment]
There is code duplication between check and ensure functions, so it is a good target for simple refactoring.

File: sentinelsat/sentinel.py [hunk 1]
```diff
@@ -840,6 +840,15 @@ def ensure_2d(geometry):
...
         else:
             return geometry[:2]
+        
+    def check_bounds(geometry):
+        if isinstance(geometry[0], (list, tuple)):
+            return list(map(check_bounds,geometry))
+        else:
+            if geometry[0]>180 or geometry[0]<-180:
+                raise ValueError('Longitude is out of bounds, check your JSON format or data')
+            if geometry[1]>180 or geometry[1]<-180:
```
Comment (j08lue) [inline sentinelsat/sentinel.py]: Latitude should be within `[-90, 90]`...

File: sentinelsat/sentinel.py [hunk 2]
```diff
@@ -840,6 +840,15 @@ def ensure_2d(geometry):
             return list(map(ensure_2d, geometry))
         else:
             return geometry[:2]
+        
+    def check_bounds(geometry):
+        if isinstance(geometry[0], (list, tuple)):
+            return list(map(check_bounds,geometry))
```
Comment (j08lue) [inline sentinelsat/sentinel.py]: Please PEP8-format your code.

https://github.com/sentinelsat/sentinelsat/blob/master/CONTRIBUTE.rst#code-conventions

**Comment by j08lue** [comment]
Since this, in a way, is a new feature, can you please add a new test, too?

https://github.com/sentinelsat/sentinelsat/blob/master/CONTRIBUTE.rst#code-conventions

**Comment by Andrey-Raspopov** [comment]
I'm sorry for this mess with commits, was trying to Verify ones that I've pushed with local git, but failed. Anyway, I've made changes that you asked.

**Comment by j08lue** [comment]
Very nice, @Andrey-Raspopov, thanks.

> I'm sorry for this mess with commits

No worries. I'll just squash the commits into one.

**Comment by j08lue** [comment]
Oops, I forgot about the CHANGELOG. Here is your chance for yet another pull request, @Andrey-Raspopov. 😉 Will you do it?

**Comment by Fernerkundung** [comment]
Changelog and Author list are populated with #241

**Pull request merged**